sklearn/tests/test_discriminant_analysis.py

```diff
@@ -257,6 +260,33 @@ def test_lda_store_covariance():
     )
 
 
+@pytest.mark.parametrize('n_features', [3, 5])
+@pytest.mark.parametrize('n_classes', [5, 3])
+def test_lda_dimension_warning(n_classes, n_features):
+    RNG = check_random_state(0)
+    n_samples = 10
+    X = RNG.randn(n_samples , n_features)
+    # we create n_classes labels by repeating and truncating a
+    # range(n_classes) until n_samples
+    y = np.tile(range(n_classes), n_samples // n_classes + 1)[:n_samples]
+    max_components = min(n_features, n_classes - 1)
+
+    for n_components in [max_components - 1, None, max_components]:
+        # if n_components < min(n_classes - 1, n_features), no warning
+        lda = LinearDiscriminantAnalysis(n_components=n_components)
+        assert_no_warnings(lda.fit, X, y)
+
+    for n_components in [max_components + 1,
+                         max(n_features, n_classes - 1) + 1]:
```

Comment:
I don't understand the second one?

Reply:
I am not sure about this one indeed, it's just that since I test something just one unit higher than `max_components`, I thought I could test something that is higher than both `n_features` and n_classes - 1 to ensure the test works for any value of `n_components`

Reply:
That's good for me. Maybe a small comment that explains this.

Reply:
Done